chirp-impl/src/main/java/sample/chirper/chirp/impl/JpaSessionImpl.java

```diff
@@ -55,14 +65,42 @@
         });
     }
 
-    private CompletionStage<Done> close() {
+    private CompletionStage<EntityManagerFactory> createEntityManagerFactory() {
         return executeInSlickContext(() -> {
-            factory.close();
-            return Done.getInstance();
+            log.debug("Initializing JPA EntityManagerFactory with persistence unit name {}", PERSISTENCE_UNIT_NAME);
+            long startTime = System.currentTimeMillis();
+            while (System.currentTimeMillis() - startTime <= INIT_TIMEOUT_MILLIS &&
+                    !Thread.currentThread().isInterrupted()) {
+                try {
+                    return Persistence.createEntityManagerFactory(PERSISTENCE_UNIT_NAME);
+                } catch (Exception e) {
+                    log.warn("Exception while initializing JPA EntityManagerFactory", e);
+                }
+                if (System.currentTimeMillis() - startTime <= INIT_TIMEOUT_MILLIS) {
+                    log.info("Will retry initializing JPA EntityManagerFactory in {} ms", INIT_RETRY_INTERVAL_MILLIS);
+                    try {
+                        Thread.sleep(INIT_RETRY_INTERVAL_MILLIS);
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            }
+            log.error("Cancelled initializing JPA EntityManagerFactory after {} ms", INIT_TIMEOUT_MILLIS);
+            throw new CancellationException(
+                    "Cancelled initializing JPA EntityManagerFactory after " + INIT_TIMEOUT_MILLIS + " ms"
+            );
```

Comment:
I'd prefer a non blocking approach using the Akka scheduler.  It could look something like this:

```java
private CompletionStage<EntityManagerFactory> tryCreate(FiniteDuration maxDuration, FiniteDuration retryInterval) {
  CompletionStage<EntityManagerFactory> result = executeInSlickContext(() ->
    Persistence.createEntityManagerFactory(PERSISTENCE_UNIT_NAME)
  );
  if (maxDuraration.toMillis() <= 0) {
    // Duration is up, don't try and recover from failure
    return result;
  } else {
    // This wrapping and then composing with identity logic is because CompletionStage has no equivalent to Scalas Future.recoverWith
    return result.thenApply(CompletableFuture::completedFuture).exceptionally(e => {
      // Probably log the exception
      FiniteDuration timeout = retryInterval.min(maxDuration);
      return actorSystem.scheduler().scheduleOnce(timeout, () -> tryCreate(maxDuration.minus(timeout), retryInterval), actorSystem.dispatcher());
    }).thenCompose(Function.identity);
  }
}
```